### PR TITLE
Feature / default boards options mananger

### LINKED ISSUE #1311: Add a new core board based on the Picseepal 11x6 using ARASAAC symbols

### PR DESCRIPTION
**Pull request by tomivm**

### CODE REVIEW

File: src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js [hunk 1]
```diff
@@ -36,6 +36,34 @@ export default defineMessages({
...
+    defaultMessage: 'Default boards selector'
+  },
+  selectDefaultBoardTitle: {
+    id: 'cboard.components.CommunicatorToolbar.selectDefaultBoardTitle',
+    defaultMessage: 'Select a default board'
+  },
+  confirmChangeHomeBoardMessage: {
+    id: 'cboard.components.CommunicatorToolbar.confirmChangeHomeBoardMessage',
+    defaultMessage:
+      'If you accept. this action changes your home board to the selected option. You can switch it again by selecting another option.'
```
Comment (RodriSanchez1) [inline src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js]: Do you want to change your home board for the selected option? You can switch it again by selecting another option.'

File: src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+import { Card, CardMedia, Typography } from '@mui/material';
+
+import messages from '../CommunicatorToolbar.messages';
+
+const DefaultBoardOption = ({ rootBoard, onClick, intl }) => {
+  return (
+    <Card
+      sx={{
+        width: { xs: '100%' }
```
Comment (RodriSanchez1) [inline src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js]: You can add the property `cursor: 'pointer'` for better styling

File: src/reducers.js [hunk 3]
```diff
@@ -8,11 +12,26 @@ import boardReducer from './components/Board/Board.reducer';
...
+  }
+};
 
 const config = {
   key: 'root',
   storage,
-  blacklist: ['language']
+  blacklist: ['language'],
+  version: 0,
+  migrate: createMigrate(boardMigrations, { debug: true })
```
Comment (RodriSanchez1) [inline src/reducers.js]: change to  `{debug: false}`

File: src/translations/src/cboard.json [hunk 4]
```diff
@@ -207,6 +207,13 @@
   "cboard.components.CommunicatorToolbar.tiles": "{qty} Tiles",
   "cboard.components.CommunicatorToolbar.editCommunicator": "Build",
   "cboard.components.CommunicatorToolbar.addBoardButton": "Add Board",
+  "cboard.components.CommunicatorToolbar.defaultBoardsIconLabel": "Default boards selector",
+  "cboard.components.CommunicatorToolbar.selectDefaultBoardTitle": "Select a default board",
+  "cboard.components.CommunicatorToolbar.confirmChangeHomeBoardMessage": "If you accept. this action changes your home board to the selected option. You can switch it again by selecting another option.",
```
Comment (RodriSanchez1) [inline src/translations/src/cboard.json]: Update according to the new message

**Comment by RodriSanchez1** [comment]
Needs to add the prop `defaultBoardIncluded ` on the intitial state of the communicator reducer

File: src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js (hunk 2 above)
Comment (tomivm) [reply src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js]: I don't agree. Probably would be better to add a hover style.
see: https://www.youtube.com/watch?v=AnTNN_QlFvQ&t=10s

Comment (RodriSanchez1) [reply src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js]: Yeah! You can add whatever you want!

**Comment by tomivm** [comment]
> Needs to add the prop `defaultBoardIncluded ` on the intitial state of the communicator reducer

@RodriSanchez1  Are you sure that this is necessary? in principle this is intentional because the feature is designed to support old users. If the user not has default boards included and uses the feature. The communicator home board (that probably is already modified) is stored for the 'advanced' homeBoard.
Do you think that this could cause a bug?

**Comment by tomivm** [comment]
To improve the first usage of the app the default boards gallery should be integrated into the first dialog of  the boardTour
image

File: package.json [hunk 5]
```diff
@@ -20,10 +20,14 @@
...
     "@crowdin/crowdin-api-client": "^1.18.2",
     "@ctrl/react-adsense": "^1.5.0",
+    "@emotion/react": "^11.10.5",
+    "@emotion/styled": "^11.10.5",
     "@material-ui/core": "^4.12.1",
     "@material-ui/icons": "^4.11.2",
     "@material-ui/lab": "4.0.0-alpha.57",
     "@microsoft/applicationinsights-web": "^2.8.4",
+    "@mui/icons-material": "^5.11.0",
+    "@mui/material": "^5.11.0",
```
Comment (martinbedouret) [inline package.json]: We should not use the mui packages until we update the material UI version on cboard.

File: src/components/Board/Board.messages.js [hunk 6]
```diff
@@ -175,6 +175,10 @@ export default defineMessages({
     defaultMessage:
       'Here you can access your communicator, edit it and enrich it with more boards.'
   },
+  walkthroughDefaultBoardsSelector: {
+    id: 'cboard.components.Board.walkthroughDefaultBoardsSelector',
+    defaultMessage: 'Here you can switch between different default boards.'
+  },
```
Comment (martinbedouret) [inline src/components/Board/Board.messages.js:185]: change to `Here you can switch between different featured boards`

File: src/components/Communicator/Communicator.constants.js [hunk 7]
```diff
@@ -9,6 +9,10 @@ export const DELETE_BOARD_COMMUNICATOR =
   'cboard/Communicator/DELETE_BOARD_COMMUNICATOR';
 export const REPLACE_BOARD_COMMUNICATOR =
   'cboard/Communicator/REPLACE_BOARD_COMMUNICATOR';
+export const PUSH_DEFAULT_BOARD_INCLUDED =
```
Comment (martinbedouret) [inline src/components/Communicator/Communicator.constants.js]: not clear what is `PUSH` in this context. Shouldn't be `ADD`?

File: src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js [hunk 8]
```diff
@@ -36,6 +36,34 @@ export default defineMessages({
   addBoardButton: {
     id: 'cboard.components.CommunicatorToolbar.addBoardButton',
     defaultMessage: 'Add Board'
+  },
+  defaultBoardsIconLabel: {
+    id: 'cboard.components.CommunicatorToolbar.defaultBoardsIconLabel',
+    defaultMessage: 'Default boards selector'
```
Comment (martinbedouret) [inline src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js]: change to `Select a featured board`

File: src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js [hunk 9]
```diff
@@ -36,6 +36,34 @@ export default defineMessages({
...
+  },
+  selectDefaultBoardTitle: {
+    id: 'cboard.components.CommunicatorToolbar.selectDefaultBoardTitle',
+    defaultMessage: 'Select a default board'
```
Comment (martinbedouret) [inline src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js]: change to `Select a featured board`

File: src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js [hunk 10]
```diff
@@ -36,6 +36,34 @@ export default defineMessages({
...
+  },
+  confirmChangeHomeBoardMessage: {
+    id: 'cboard.components.CommunicatorToolbar.confirmChangeHomeBoardMessage',
+    defaultMessage:
+      'Do you want to change your home board for the selected option? You can switch it again by selecting another option.'
```
Comment (martinbedouret) [inline src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js]: change to `Do you want to change your current board for the selected option? You can switch it again`

File: src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js [hunk 11]
```diff
@@ -36,6 +36,34 @@ export default defineMessages({
...
+    id: 'cboard.components.CommunicatorToolbar.agree',
+    defaultMessage: 'agree'
+  },
+  close: {
+    id: 'cboard.components.CommunicatorToolbar.close',
+    defaultMessage: 'Close'
+  },
+  defaultBoardImageAlt: {
+    id: 'cboard.components.CommunicatorToolbar.defaultBoardImageAlt',
+    defaultMessage: 'Default board caption'
```
Comment (martinbedouret) [inline src/components/Communicator/CommunicatorToolbar/CommunicatorToolbar.messages.js]: Change to `Board caption`

File: src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js [hunk 12]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { CardContent } from '@material-ui/core';
+
+import { Card, CardMedia, Typography } from '@mui/material';
```
Comment (martinbedouret) [inline src/components/Communicator/CommunicatorToolbar/DefaultBoardSelector/DefaultBoardOption.js]: use the `material-ui` package

**Review by martinbedouret — CHANGES_REQUESTED** [review]

File: src/components/Communicator/Communicator.constants.js (hunk 7 above)
Comment (tomivm) [reply src/components/Communicator/Communicator.constants.js]: yes, it should be ADD! thanks

**Review by martinbedouret — APPROVED** [review]

**Pull request merged**